### PR TITLE
Allow Both Blocks and Multiple for Multi-Select

### PR DESCRIPTION
**Pull request by Sathyan08**
The fluid select is not currently well set up to pass blocks to the fluid-checkbox used in a multi-select in a fluid-select.  We will need to customize the labels for a select that we are converting to a multi-select, and we need to make that process easier in case we do that again in the future.

### CODE REVIEW

File: addon/components/fluid-checkbox.js [hunk 1]
```diff
@@ -1,3 +1,3 @@
-import templateOnly from '@ember/component/template-only';
+import Component from '@ember/component';
 
-export default templateOnly();
+export default class FluidCheckbox extends Component {}
```
Comment (nicksteffens) [inline addon/components/fluid-checkbox.js:3]: **question:** Why the change from template only if there is no logic in the js?

**Review by nicksteffens — COMMENTED** [review]
**thought(nitpick):** Instead of trying to get 1 component to do everything we create a new component for multiple select called `<FluidSelect::Multiple .../>` to do exactly what we need it to do. If it needs to be on multiple packages (aka Canvas, Hogwarts) it should live in fantastic beasts and could extend original `FluidSelect`. Otherwise if it does need to belong under the design system, it should be a new component as listed above.

https://www.notion.so/movableinkepd/movable-fluid-ec6668b4194448fdbb4405dce3f54b64?pvs=4#1b8c2a862af54637a97ffe39d8403811

File: addon/components/fluid-checkbox.js (hunk 1 above)
Comment (Sathyan08) [reply addon/components/fluid-checkbox.js]: I am probably going to revert this once

> **thought(nitpick):** Instead of trying to get 1 component to do everything we create a new component for multiple select called `<FluidSelect::Multiple .../>` to do exactly what we need it to do. If it needs to be on multiple packages (aka Canvas, Hogwarts) it should live in fantastic beasts and could extend original `FluidSelect`. Otherwise if it does need to belong under the design system, it should be a new component as listed above.
> 
> https://www.notion.so/movableinkepd/movable-fluid-ec6668b4194448fdbb4405dce3f54b64?pvs=4#1b8c2a862af54637a97ffe39d8403811

I think it does makes sense to create a separate multi-select component.  The extensive use of `else if` is a pretty big code smell here which I think suggests a need for a new component.  For now, I am going to focus on getting the actual data source multi-select working and see how the code looks when I actually use apply it in our front-end.  Once I am good there, I will probably try to refactor this with your feedback in mind.

File: addon-test-support/pages/fluid-select.js [hunk 2]
```diff
@@ -14,6 +14,8 @@ import { assert } from '@ember/debug';
 export const FluidSelect = {
   scope: '.fluid-select',
 
+  // TODO: we should rename this because this function both opens and closes multi-select dropdowns
+  // we should just rename it something like 'toggle'
```
Comment (nicksteffens) [inline addon-test-support/pages/fluid-select.js:18]: 👍

**Review by nicksteffens — APPROVED** [review]
We'll break this out at a later date, so that we don't have 1 component do ALL the things.
